gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -0,0 +1,264 @@
+package com.google.gcloud.storage.testing;
+
+import com.google.api.services.storage.model.Bucket;
+import com.google.api.services.storage.model.StorageObject;
+import com.google.gcloud.spi.ServiceRpcFactory;
+import com.google.gcloud.spi.StorageRpc;
+import com.google.gcloud.storage.StorageException;
+import com.google.gcloud.storage.StorageOptions;
+
+import sun.misc.IOUtils;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.NoSuchFileException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A bare-bones in-memory implementation of Storage, meant for testing.
+ */
+public class FakeStorageRpc implements StorageRpc {
```

Comment:
The design of this class is suboptimal because it's working around technical debt on my previous design. JP and I already had a similar discussion offline. In a follow-up PR, I plan on refactoring the way GCS NIO classes get instantiated so this fake can be used more elegantly. They'll also be more injection friendly.